docs/content/Schema/Fundamentals/Concepts.md

```diff
@@ -0,0 +1,323 @@
+as colors, URLs or SKUs.
+
+[The `orders` table](#top) contains only dimensions, so representing them in the
+`Orders` cube is straightforward:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  dimensions: {
+    id: {
+      sql: `id`,
+      type: `number`,
+      // Here we explicitly let Cube.js know this field is the primary key
+      // This is required for de-duplicating results when using joins
+      primaryKey: true
+    },
+    status: {
+      sql: `status`,
+      type: `string`
+    },
+  },
+});
+```
+
+[The `line_items` table](#top) also has a couple of dimensions which can be
+represented as follows:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+    id: {
+      sql: `id`,
+      type: `number`,
+      // Again, we explicitly let Cube.js know this field is the primary key
+      // This is required for de-duplicating results when using joins
+      primaryKey: true
+    },
+  },
+});
+```
+
+Dimensions can be of different types, and you can find them all
+[here][ref-schema-dimension-types].
+
+### Time Dimensions
+
+<!-- prettier-ignore-start -->
+[[info | ]]
+| Time-based properties are generally used as dimensions in Cube.js. // say why
+<!-- prettier-ignore-end -->
+
+Time-based properties should be represented as dimensions with type `time`. Time
+dimensions allow granulating the result set for queries sent to Cube.js.
+
+We can add the necessary time dimensions to both schemas as follows:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+
+    completedAt: {
+      sql: `completed_at`,
+      type: `time`,
+    },
+  },
+});
+```
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+  },
+});
+```
+
+Time dimensions are essential to enabling performance boosts such as
+[partitioned pre-aggregations][ref-caching-use-preaggs-partition-time] and
+[incremental refreshes][ref-tutorial-incremental-preagg].
+
+## Measures
+
+Measures are measurable, or [quantitative properties][wiki-quantitative] of the
+cube; they are often numeric and mathematical operations such as `SUM()` or
+`MIN()` **can** be performed on them. This also means that measures are expected
+to [be additive][ref-caching-additive].
+
+To add a measure called `count` to our `Orders` cube, for example, we can do the
+following:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  measures: {
+    count: {
+      type: `count`,
+    },
+  },
+});
+```
+
+In our `LineItems` cube, we can We could also create a measure to sum up the
+total value of line items sold:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  measures: {
+    total: {
+      sql: `price`,
+      type: `sum`,
+    },
+  },
+})
+```
+
+> Add a mention of how to create measures from dimensions (SUM something?)
```

Comment:
@keydunov Is it worth talking about converting dimensions to measures here or should we do that in advanced concepts/recipes instead?